src/Workspace.cpp

```diff
@@ -13,9 +17,12 @@ void WorkspaceFolder::openTextDocument(const lsp::DocumentUri& uri, const lsp::D
     fileResolver.managedFiles.emplace(
         std::make_pair(normalisedUri, TextDocument(uri, params.textDocument.languageId, params.textDocument.version, params.textDocument.text)));
 
-    // Mark the file as dirty as we don't know what changes were made to it
-    auto moduleName = fileResolver.getModuleName(uri);
-    frontend.markDirty(moduleName);
+    if (isConfigured)
```

Comment:
I believe it was done because `getModuleName` now might involve accessing `platform`.

Reply:
Cool. I wonder if this can cause any problems where we didn't mark a file dirty appropriately on startup, but I think it is fine for now. I want to get rid of the reliance on platform-specific file paths across the server except for file resolution, so hopefully this won't be a problem in future. Will note it down to resolve: #614